### PR TITLE
get dependencies of bundles

### PR DESCRIPTION
**Pull request by erwinmombay**
extensions are just currently hardcoded. I'd like to ideally extract the `declareExtension` values into a config file that both this code and the code in gulpfile.js can read.

code in get-dep-graph.js is code extracted from splittable project and modified to work with our code base.

(see also previously approved PR https://github.com/ampproject/amphtml/pull/12975)

### CODE REVIEW

File: build-system/base.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @const {!Object}  */
+const process = {};
+
+process.env = {};
+
+process.env.NODE_ENV = 'production';
```
Comment (rsimha) [inline build-system/base.js]: Is there any reason the above three lines can't be combined into one line like this?
```
const process = {
  env: {
    'NODE_ENV': 'production',
  },
};
```

Comment (erwinmombay) [reply build-system/base.js]: this is an extern file so we can't really write it like normal javascript

**Comment by erwinmombay** [comment]
@rsimha this is a constant moving target since im experimenting with stuff to fix the large _base.js, but id like to get something merged

File: build-system/base.js (hunk 1 above)
Comment (erwinmombay) [reply build-system/base.js]: oh wait this is just base.js my bad. yeah we can write it that way

**Comment by rsimha** [comment]
LGTM alerts for unused warnings have been disabled. You can ignore https://github.com/ampproject/amphtml/pull/15969#issuecomment-399176602

**Pull request merged**